socialhome/streams/app/components/StreamElement.vue

```diff
@@ -88,6 +93,9 @@ export default Vue.component("stream-element", {
         showShares() {
             return this.isUserAuthenticated || this.sharesCount > 0
         },
+        getHasShared() {
```

Comment:
I'm not fan of the `get`. These are supposed to be attributes and are accessed like ones in the generated Vue code. Maybe `postWasShared`?

Reply:
I'm not really a fan of how this works at all :( It's annoying to have to specify 3 things just to get one variable to mutate. `postWasShared` doesn't indicate that we're only talking about the current browsing user. I'll read a bit about local state management in Vue, there must be a better way than to have to do this way :/

Reply:
Well, there doesn't seem to be a way, I guess we talked about this earlier. I think the `_` data was a mistake, it forces to use `computed` to get the actual value in template.

How about one of the three naming conventions, for local data that uses props as initial values. As an example, `hasShared` prop, becomes mutable as:

* `localHasShared`
* `hasSharedLocal`
* `hasShared$`

I kind of like the last one, since it doesn't rely on the word `local` which could have other meanings. If not, I prefer postfixing with `local`, instead of prefixing.

Opinions / other options?

Reply:
> I'm not really a fan of how this works at all

I know. Unhopefully, Vue components have no constructors like React have. So no way to play with props during construction. There must be a right way. I assume we are not the firsts to run into this use case. I think the whole component must be refactored to use a store anyway. So, for now, `hasShared$` is ok.

Reply:
Actually, shouldn't the store be an object in the first place, not an array? This way component could just update count of shares in the store instead of maintaining a local mutable copy.. But that is out of scope of this PR, maybe when the API is done refactor the store to be an object so content can be referenced from components by their id.

Reply:
> This way component could just update count of shares in the store instead of maintaining a local mutable copy..

Yes, that is my idea. But we're still struggling to find the correct way. No need for early optimisations. It works. That's all that matters for now. I'll refactor later. Please merge so I can rebase #273 ;)